content/5_Plat/Convex_Hull.mdx

```diff
@@ -267,7 +267,192 @@ int main()
 
 ### Solution
 
-<IncompleteSection />
+With the Monotone Chain algorithm, we start by sorting the given $n$ points in ascending order with respect to their $x$ coordinates. If two points have the same $x$ value, then we will look at the $y$ coordinate. 
+
+Next, we will calculate the convex hull in two parts - the upper and the lower hull. Firstly, we observe that the starting and ending points of both upper and lower hulls are the same. They are the points with the lowest and highest $x$ value, $P_0$ and $P_{n-1}$. We primarily add $P_0$ and $P_1$ to the hull. Note that $P_1$ doesn't necessarily have to be a point of the convex hull at the end. Then, starting from $P_2$, we iterate through the sorted points and add them to the hull. Let's denote the current point being inspected as $P_{k}$ and the last point in the hull as $P_{i}$. While adding new points, we want to make sure that there is no right turn among all segments of the hull, just like in the Graham Algorithm discussed above. To achieve this, the segment $P_{i-1}P_i$ should always be on the left side of the segment $P_{i-1}P_k$. This can be calculated by a cross product trick: 
+
+- If $(P_i - P_{i-1}) \times (P_k - P_{i-1}) < 0$, the point $P_i$ lies on the left side of the segment $P_{i-1}P_k$. In this case, we have to remove point $P_i$ from the hull and repeat this check. 
+- If $(P_i - P_{i-1}) \times (P_k - P_{i-1}) = 0$, the point $P_i$ lies on the segment $P_{i-1}P_k$. Whether to include multiple collinear points depends on the question, but to the given question above, we will remove the point $P_i$ as well and repeat the check. 
+- Otherwise, the point $P_i$ lies on the right side of the segment $P_{i-1}P_k$. So we are good to inspect the next point $P_{k+1}$ from the given point list. 
+
+After all points being examined, we have calculated the lower hull and will begin to find the upper hull in the same manner. This time, we add point $P_{n-2}$ to the hull and iterate from the end of the points, $P_{n-3}$, to the starting point $P_0$. The Point $P_{n-2}$ doesn't have to be the convex hull and could be removed if it causes a right turn. 
+
+At the end, we have traversed the points in the counterclockwise order. To do this in the clockwise order, one only has to change the condition for $(P_i - P_{i-1}) \times (P_k - P_{i-1})$ from more than 0 to less than 0. In this case, the upper hull will be found first and then the lower hull. 
```

Comment:
> At the end, we have traversed the points in the counterclockwise order.

We have? Do you rather mean we've found the points on the convex hull in counterclockwise order?

Reply:
ya we got the points of the hull in counterclockwise order, not the traversal. Im gonna fix it.